app/src/features/apiClient/screens/apiClient/components/modals/MoveToCollectionModal/MoveToCollectionModal.tsx

```diff
@@ -0,0 +1,118 @@
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+import { toast } from "utils/Toast";
+import { RQButton } from "lib/design-system/components";
+import { redirectToRequest } from "utils/RedirectionUtils";
+import { useNavigate } from "react-router-dom";
+import "./moveToCollectionModal.scss";
+
+interface Props {
+  recordToMove: RQAPI.Record;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const MoveToCollectionModal: React.FC<Props> = ({ isOpen, onClose, recordToMove }) => {
+  const navigate = useNavigate();
+  const { apiClientRecords, onSaveRecord } = useApiClientContext();
+  const [selectedCollection, setSelectedCollection] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const user = useSelector(getUserAuthDetails);
+  const teamId = useSelector(getCurrentlyActiveWorkspace);
+
+  const collectionOptions = useMemo(() => {
+    return apiClientRecords
+      .filter((record) => record.type === RQAPI.RecordType.COLLECTION)
+      .map((record) => ({
+        label: record.name,
+        value: record.id,
+      }));
+  }, [apiClientRecords]);
+
+  const handleRecordMove = useCallback(async () => {
+    setIsLoading(true);
+    if (selectedCollection?.__isNew__) {
+      //create new collection and move request to it
+      const collectionToBeCreated: Partial<RQAPI.CollectionRecord> = {
+        collectionId: "",
+        name: selectedCollection.label,
+        type: RQAPI.RecordType.COLLECTION,
+        deleted: false,
+        data: {},
+      };
+      setIsLoading(true);
+      const newCollection = await upsertApiRecord(user?.details?.profile?.uid, collectionToBeCreated, teamId);
+      if (newCollection.success) {
+        onSaveRecord(newCollection.data);
+        const updatedRequest = { ...recordToMove, collectionId: newCollection.data.id };
+        const result = await upsertApiRecord(user?.details?.profile?.uid, updatedRequest, teamId);
+        if (result.success) {
+          toast.success("Request moved to collection successfully");
+          onSaveRecord(result.data);
+          redirectToRequest(navigate, result.data.id);
+        } else {
+          toast.error("Failed to move request to collection");
+        }
+      } else {
+        toast.error("Failed to create a new collection");
+      }
+      setIsLoading(false);
+      onClose();
+    } else {
+      //move request to existing collection
+      const updatedRequest = { ...recordToMove, collectionId: selectedCollection.value };
+      const result = await upsertApiRecord(user.details?.profile?.uid, updatedRequest, teamId);
+      setIsLoading(false);
+      if (result.success) {
+        toast.success("Request moved to collection successfully");
+        onSaveRecord(result.data);
+        redirectToRequest(navigate, result.data.id);
+      } else {
+        toast.error("Failed to move request to collection");
+      }
+      onClose();
+    }
```

Comment:
1. wrap inside a try-catch since `upsertApiRecord` is only bubbling errors and not handling them
2. move `setIsLoading(false); onClose();` to the end, outside the `if-else`